src/drivers/libraries/SAL.qll

```diff
@@ -0,0 +1,224 @@
+/**
+ * Provides classes for identifying and reasoning about Microsoft source code
+ * annotation language (SAL) macros.
+ * 
+ * This version of this file has been updated to include driverspecs.h as a valid SAL header.
+ * It will be removed if and when that change is upstreamed to the main CodeQL repo.
+ * 
+ * TODO: Create PR to bring into main CodeQL repo or recieve feedback on how to more cleanly
+ * extend this module in our own file without copying the existing library.
+ */
+
+import cpp
+
+/**
+ * A SAL macro defined in `sal.h` or a similar header file.
+ */
+class SALMacro extends Macro {
+  SALMacro() {
+    this.getFile().getBaseName() =
+      ["sal.h", "specstrings_strict.h", "specstrings.h", "w32p.h", "minwindef.h", "driverspecs.h"] and
+    (
+      // Dialect for Windows 8 and above
+      this.getName().matches("\\_%\\_")
+      or
+      // Dialect for Windows 7
```

Comment:
Are we supporting anything before Windows 7?

Reply:
I doubt it (and I'm actually unsure if SAL was available for public use before Windows 7 - at the very least Win7 was where we first introduced PreFAST for Drivers, which used SAL for driver analysis).  In any event this code is just taken 1:1 from the main CodeQL repo.

Reply:
We should not support Windows 8.1, Windows 8, or Windows 7.  This to follow the style of the newest WDK.